### PR TITLE
[LOOP-1993] Workout override reminder

### PR DESCRIPTION
**Pull request by nhamming**
https://tidepool.atlassian.net/browse/LOOP-1993

Remind the user if the workout override has been enabled for more than 24 hours. 

Note: 
- This work will be refactored with https://tidepool.atlassian.net/browse/LOOP-2140

### CODE REVIEW

File: LoopCore/LoopSettings.swift [hunk 1]
```diff
@@ -301,3 +319,41 @@ extension LoopSettings: RawRepresentable {
...
+    public var workoutOverrideReminderAlert: Alert {
+        let title = NSLocalizedString("Workout Temp Adjust Still On", comment: "Workout override still on reminder alert title")
+        let body = NSLocalizedString("Workout Temp Adjust has been turned on for more than 24 hours. Make sure you still want it enabled, or turn it off in the app.", comment: "Workout override still on reminder alert body.")
+        let content = Alert.Content(title: title,
+                                    body: body,
+                                    acknowledgeActionButtonLabel: NSLocalizedString("Dismiss", comment: "Default alert dismissal"))
+        return Alert(identifier: workoutOverrideReminderAlertIdentifier,
+                     foregroundContent: content,
+                     backgroundContent: content,
+                     trigger: .repeating(repeatInterval: TimeInterval.minutes(1)))// Just for Dev testing. should be TimeInterval.hours(24)))
```
Comment (rickpasetto) [inline LoopCore/LoopSettings.swift]: Change to `.hours(24)`?

File: LoopCore/LoopSettings.swift [hunk 2]
```diff
@@ -213,6 +223,14 @@ extension LoopSettings {
         }
 
         guard let override = scheduleOverride else { return }
```
Comment (rickpasetto) [inline LoopCore/LoopSettings.swift]: [nit] consider another variable name other than `override` (which is a Swift keyword)

**Review by rickpasetto — APPROVED** [review]
LGTM (1 non-blocking comment, one reminder to change the time interval)

File: LoopCore/LoopSettings.swift (hunk 1 above)
Comment (nhamming) [reply LoopCore/LoopSettings.swift]: Remember, this is just an initial review.

File: LoopCore/LoopSettings.swift (hunk 2 above)
Comment (nhamming) [reply LoopCore/LoopSettings.swift]: Good point. This was the original variable name, but I'll definitely update it here (and throughout that file).

File: Loop/Managers/Alerts/AlertManager.swift [hunk 3]
```diff
@@ -179,6 +179,20 @@ extension AlertManager {
...
+            case .success(let alerts):
+                alerts.forEach { alert in
+                    do {
+                        self.replayAlert(try Alert(from: alert, adjustedForStorageTime: true))
+                    } catch {
+                        self.log.error("Error decoding alert from persistent storage: %@", error.localizedDescription)
+                    }
+                }
+            }
+        }
```
Comment (nhamming) [inline Loop/Managers/Alerts/AlertManager.swift:195]: @rickpasetto While this works for this use case, I'm curious if there are other recurring alerts that should not be rescheduled on launch after they are acknowledged?

File: Loop/Managers/Alerts/UserNotificationAlertPresenter.swift [hunk 4]
```diff
@@ -38,10 +38,8 @@ class UserNotificationAlertPresenter: AlertPresenter {
     }
     
     func retractAlert(identifier: Alert.Identifier) {
-        DispatchQueue.main.async {
-            self.userNotificationCenter.removePendingNotificationRequests(withIdentifiers: [identifier.value])
-            self.userNotificationCenter.removeDeliveredNotifications(withIdentifiers: [identifier.value])
-        }
+        self.userNotificationCenter.removePendingNotificationRequests(withIdentifiers: [identifier.value])
+        self.userNotificationCenter.removeDeliveredNotifications(withIdentifiers: [identifier.value])
```
Comment (nhamming) [inline Loop/Managers/Alerts/UserNotificationAlertPresenter.swift]: Not sure what this was on `DispatchQueue.main`. Seems unnecessary. Am I overlooking something?

File: Loop/AppDelegate.swift [hunk 5]
```diff
@@ -129,6 +129,9 @@ final class AppDelegate: UIResponder, UIApplicationDelegate, DeviceOrientationCo
 
     func applicationWillTerminate(_ application: UIApplication) {
         log.default(#function)
+        
+        // if the app is being terminated, remove any workout override reminders
+        alertManager.retractAlert(identifier: LoopSettings.workoutOverrideReminderAlertIdentifier)
```
Comment (rickpasetto) [inline Loop/AppDelegate.swift]: I would remove this.  If the user has left this override turned on, why should it matter that the app is still running to get this reminder?

File: Loop/Managers/Alerts/AlertManager.swift (hunk 3 above)
Comment (rickpasetto) [reply Loop/Managers/Alerts/AlertManager.swift:195]: I don't think you need this.  If it is acknowledged, it shouldn't play back.  I need to understand your use case more.

**Review by rickpasetto — CHANGES_REQUESTED** [review]
@nhamming Thanks for proposing these changes, but I need to chat with you more live to understand your changes.

File: Loop/Managers/Alerts/UserNotificationAlertPresenter.swift (hunk 4 above)
Comment (rickpasetto) [reply Loop/Managers/Alerts/UserNotificationAlertPresenter.swift]: IIRC operations on UNUserNotificationCenter need to occur on `main`.  Have you tested it to make sure if it happens on another thread that there is no runtime error?  At any rate, IMHO I'd revert this change.

**Review by rickpasetto — APPROVED** [review]
Thanks for making these changes.  LGTM. 👍

**Pull request merged**